plugin/keyring_vault/vault_keys_container.cc

```diff
@@ -0,0 +1,34 @@
+#include <my_global.h>
+#include "vault_keys_container.h"
+
+namespace keyring
+{
+  my_bool Vault_keys_container::init(IKeyring_io* keyring_io, std::string keyring_storage_url)
+  {
+    vault_io = dynamic_cast<IVault_io*>(keyring_io);
```

Comment:
It is not a hack, it is just for DBUG_ASSERT to make sure only Vault_io is passed to this function. I have put it all into DBUG_ASSERT statement to make it more obvious.